node_inc.ks

```diff
@@ -0,0 +1,50 @@
+// Change inclination by planning a burn at the ascending or
+// descending node, whichever comes first.
+
+// TODO: Consider dV for both nodes and maybe select the more efficient (when orbit:eccentricity is high)
+
+// Desired orbital inclination,
+// default is zero or match target if selected
+parameter target_inclination is "".
+
+runOncePath("lib_util").
+utilRemoveNodes().
+
+local di is 0. // inclination difference
+local dt is 0. // time to DN
+local d2 is 0. // time to AN
+local t0 is time:seconds.
+
+if hasTarget and target_inclination = ""
+{
+	local sp is ship:position-body:position.
+	local sv is ship:velocity:orbit.
+	local tp is target:position-body:position.
+	local tv is target:velocity:orbit.
+	local sn is vcrs(sv, sp). // normal vector
+	local tn is vcrs(tv, tp).
+	local ln is vcrs(sn, tn). // from body to DN
+
+	set di to vang(sn, tn).
+	set dt to utilDtTrue(orbit:trueAnomaly+180-vang(sp, ln)).
+	set d2 to utilDtTrue(orbit:trueAnomaly+180+vang(sp,-ln)).
+}
```

Comment:
I am currently testing
```
		set di to -vang(sn, tn).
		set a1 to vang(sp, ln).
		set dt to dtTrue(orbit:trueAnomaly+a1).
		set d2 to dtTrue(orbit:trueAnomaly-a1).
```
instead and it works most of the time, but I still found a cornercase - two highly inclined and elliptical orbits: it does not seem to be problem of `dt` calculation but rather `di` and the later `sin` and `cos` as it hits AN node correctly, but `dv` of the node is wrong. The core of the problem could be that `vang` does not give a sign (could use some `arctan2` + normal vector to get proper sign) but I am stuck with this right now, probably need a break :)

Reply:
I think I know what is wrong: RETROGRADE orbits! vang is never negative, inclination is never negative, that causes the problems, I have to use plane-normal vectors. My brain just needs more time to process it in the background :)

Reply:
Haha! Yes; you can use `target:obt` parameters to spot cases where the target is retrograde to you. (Of course, if he's retrograde, then matching planes with him doesn't do much good unless you change inclination by 180+D degrees!)

Regarding your idea for node execution via delegates -- that is intriguing. I have noticed that the "certainty" of node planning gets very low when you plan a node far in the future, or with a very precise dv. Passing a delegate to the node script would be an elegant way to fix that.

I also notice you have begun using a `lib/` subdir which is something I was considering doing. The kOS scripts are so large now that it's impossible to install them at boot -- and most missions will never require most scripts. I was thinking that a simple solution would be to always copy `lib/` (and trim down the contents to ensure the code is really reusable and DRY), then make the mission script copy the specific things it knows that it will need.

Reply:
Almost everything would have to be rewritten to functions in order to use delegates, because you cannot create delegate from a script, only from function. I will test compiling libraries and review booting, but I first want scripts that really work and do the job. I am not worried about loosing signal right now, I usually take many satellite contracts and add some relay on it.